### DISCUSSION TITLE
How to confirm the instrumentation module version?

### DISCUSSION THREAD

**Discussion by zenvzenv** [open-telemetry/opentelemetry-java-instrumentation#9405] [Q&A] [answered]
There are so many insturmantation module in the project,but some instrumentation module has many version,for example:tomcat has `tomcat-7.0` and `tomcat-10.0`.

I would like to know how to distinguish between versions.Thanks.

**Comment by mateuszrzeszutek** [comment]
Hey @zenvzenv ,

> I would like to know how to distinguish between versions.Thanks.

I'm not really sure what you mean by that.

**Reply by zenvzenv** [reply]
I see that in the project structure, there are many versions of project directories under the instrumentation module. For example, there are tomcat7 and tomcat10 in the tomcat directory, which means that opentelemetry-javaagent supports tomcat7 and tomcat10. How does opentelemetry-javaagent distinguish between them? There are two versions of tomcat.
The directory structure is as follows:
```text
tomcat
  tomcat-7.0
    javaagent
  tomcat-10.0
    javaagent
```

**Reply by laurit** [reply, marked as answer]
Agent decides whether to use tomcat 7 or 10 instrumentation based on the presence of servlet5 api.
